### PR TITLE
auth: add implicit auth snippet

### PR DESCRIPTION
**Pull request by rakyll**

### CODE REVIEW

**Comment by broady** [comment]
This is only good for those style libraries. Do we not care about google.golang.org/api/... packages at all?

This sample may actually add to the confusion.

File: auth/snippets.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Package snippets contain Google Cloud authentication snippets.
+package snippets
```
Comment (broady) [inline auth/snippets.go]: auth_snippets?

File: auth/snippets.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Package snippets contain Google Cloud authentication snippets.
```
Comment (broady) [inline auth/snippets.go]: s/contain/contains/

File: auth/snippets.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	"cloud.google.com/go/storage"
+	"golang.org/x/net/context"
+	"google.golang.org/api/iterator"
+)
+
+func adc() {
+	ctx := context.Background()
+
+	// [START auth_cloud_implicit]
+	// Instantiate a client. If you don't specify credentials
```
Comment (broady) [inline auth/snippets.go]: Get rid of "instantiate". We're not Java.

File: auth/snippets.go (hunk 1 above)
Comment (rakyll) [inline auth/snippets.go]: underscore in package names is a lint error, used authsnippets instead.

File: auth/snippets.go (hunk 2 above)
Comment (rakyll) [inline auth/snippets.go]: Done

File: auth/snippets.go (hunk 3 above)
Comment (rakyll) [inline auth/snippets.go]: Done

**Pull request merged**